### ISSUE TITLE
Timezone issue on Python 3

### ISSUE THREAD

**Issue by benjaminrigaud** [simon-weber/python-libfaketime#43]
Hello,

The following test is failing on Ubuntu & Python 3.6.4 & Pytest:

``` python

def test_libfaketime():
    now = datetime.now()

    with fake_time(now):
        new_now = datetime.now()
    assert abs((new_now - now).total_seconds()) < 2

    another_now = datetime.now()
    assert abs((another_now - now).total_seconds()) < 2

```

with the stacktrace:

``` python
>       assert abs((another_now - now).total_seconds()) < 2
E       assert 7199.9999 < 2
E        +  where 7199.9999 = abs(-7199.9999)
E        +    where -7199.9999 = <built-in method total_seconds of datetime.timedelta object at 0x7fb9173b4210>()
E        +      where <built-in method total_seconds of datetime.timedelta object at 0x7fb9173b4210> = (datetime.datetime(2018, 4, 12, 13, 0, 0, 22061) - datetime.datetime(2018, 4, 12, 15, 0, 0, 21961)).total_seconds

```

It looks like the contextmanager is not configuring the time (or timezone?) back.

It works fine with libfaketime==0.4.4

**Comment by benjaminrigaud** [comment]
Ok, I'll try to reproduce a sandbox env ASAP :)
Thank you.

**Comment by benjaminrigaud** [comment]
I think the difference is I am on Europe/Paris timezone.

> date
lundi 16 avril 2018, 13:49:21 (UTC+0200)

It works fine with `TZ=UTC pytest`

I tried to reproduce a sandbox env with docker, but am stuck at the timezone part. I'll dig a bit more this part.

**Comment by benjaminrigaud** [comment]
It looks like the issue is related to a custom python build (with pew & pythonz).

**Comment by azmeuk** [comment]
I can confirm the bug when I use `env TZ=UTC+1 pytest`, and everything goes fine when I use `env TZ=UTC pytest`.

**Comment by simon-weber** [comment]
I'm back from vacation! Nice find. I should have a chance to dig into this over the weekend.

**Comment by simon-weber** [comment]
Huh, still no luck for me:
```
(venv-3) ~/python-libfaketime$ git diff
diff --git a/test/test_faketime.py b/test/test_faketime.py
index 4c8b44b..72b6ff9 100644
--- a/test/test_faketime.py
+++ b/test/test_faketime.py
@@ -131,3 +131,14 @@ class TestClassDecorator:
     @fake_time('2001-01-01')
     def test_overwrite_with_func_decorator(self):
         assert datetime.datetime(2001, 1, 1) == datetime.datetime.now()
+
+
+def test_libfaketime():
+    now = datetime.datetime.now()
+
+    with fake_time(now):
+        new_now = datetime.datetime.now()
+        assert abs((new_now - now).total_seconds()) < 2
+
+        another_now = datetime.datetime.now()
+        assert abs((another_now - now).total_seconds()) < 2

(venv-3) ~/python-libfaketime$ env TZ=UTC+1 pytest
re-exec with libfaketime dependencies
===================================================================== test session starts =====================================================================
platform linux -- Python 3.6.2, pytest-3.2.1, py-1.5.3, pluggy-0.4.0
rootdir: /home/simon/python-libfaketime, inifile:
collected 27 items                                                                                                                                             

test/test_faketime.py .................
test/test_freezegun.py ..........

================================================================== 27 passed in 0.30 seconds ==================================================================
```

**Comment by azmeuk** [comment]
@simon-weber What happens if you checkout [my branch](https://github.com/azmeuk/python-libfaketime/tree/issue43) and run `tox`?
[This travis job](https://travis-ci.org/azmeuk/python-libfaketime/jobs/378734384) successfully reproduces the error.

**Comment by simon-weber** [comment]
Sweet, I was able to recreate with your tox example.

This is pretty weird. When I inspect the faketime environment variables before/during/after the context manager everything looks normal on both working and broken pythons.

Digging through the python changelog I noticed only one item that seemed related: https://bugs.python.org/issue28148. My best guess right now is that there's a bug in libfaketime with the treatment of the new syscalls that python switched to.
